### ISSUE TITLE
Room Tone Control

### ISSUE THREAD

**Issue by DR7** [adelyte/crescendo#80]
I was wondering if a module exists for tone control for each room such as being able to select and save Bass,Treble,Balance as well as being able to do the same with Min/Max Volume and System Startup Volume for each room?

I looked at many of the Crestron modules and I find the "Stereo Volume and Tone Control v1.1" it doesn't seem to properly control the Bass and Treble and seems buggy.

If a person was going to attempt to create something like this is their a control ID on equipment ID that should be used? I did refer to the crosspoint routing page and to be honest I am not sure I can even pull this off or have the time and that's why I thought to ask if someone already created a module with this functionality.

I can, of course, do everything to control these values, save them to ARAM and recall them but my lack of skill is with packaging it all into a module that is a crosspoint so I can have one "Audio Settings" page used for all rooms instead of having spaghetti code with buffers and multiple pages to accomplish a simple task that's needed for all rooms.

Thanks in advance.

**Comment by adelyte-austin** [comment]
@DR7 
There is no Crescendo compatible module for non-volatile tone control since many of these settings do not need to change at runtime, like `min / max volumes`. Sonnex tool made eq controls even less important. Typically if a user requires this for a BIPAD8 I write a monolithic `Device` module: `D-05  Tone Controls` (ID is arbitrary) and the appropriate views. Basic ramp and toggle logic will work here. Use an `ARAM` and `DRAM` symbol inside the _System_ folder at the top of the program and those settings will persist.
You could also create a wrapped device module with the necessary logic for each audio output, then have the interface select the appropriate device based on the current room. This could be `D-11` through `D-18`.

The correct method would be to write a full implementation using simpl+ module that writes to file, then use the `Switcher - D4` ID pathway and connect it to the generic layer for interfaces. There is no plan for this currently, but perhaps users would like a poll to decide its importance?

**Comment by DR7** [comment]
Thank you Austin for the detailed reply, my SIMPL+ was enough to get certified and that was years ago and I've unfortunately avoided it since and really should practice. My crosspoints also are not the best but I am learning by looking inside your modules.

One thing I did notice tonight was the SWAMP offers the selection of tone profiles via analog using the EQ_Selection signal and sending it a 0-5. I think this would take care of most peoples needs but for whatever reason some users want all the adjustments while myself I just set it and forget it by logging into the SWAMP.

Other frameworks I have used do include these functions and seem to have a 100-200 or more signals in their modules(WAY to many!) and usually have an Audio Zone control module and a Media Room module but the size of the others modules I have seen are huge, Crescendo is small clean and efficient and I prefer picking all my sources using /x01/x02 etc instead of choosing to enable or disable from dropdowns in parameters.

Myself, of course, I would vote for the poll but not so sure how others feel about the importance, its one of the only things I've noticed missing from the framework so far so I thought to ask because I assumed maybe someone else had created a module already I could easily use.
